swap/src/protocol/alice.rs

```diff
@@ -168,27 +168,47 @@ impl Behaviour {
     }
 
     /// Alice always sends her messages as a response to a request from Bob.
-    pub fn send_amounts(&mut self, channel: ResponseChannel<AliceToBob>, amounts: SwapAmounts) {
+    pub fn send_amounts(
+        &mut self,
+        channel: ResponseChannel<AliceToBob>,
+        amounts: SwapAmounts,
+    ) -> Result<()> {
         let msg = AliceToBob::Amounts(amounts);
-        self.amounts.send(channel, msg);
+        let _ = self.amounts.send(channel, msg);
```

Comment:
```suggestion
        self.amounts.send(channel, msg)?;
```
Any reason why you silence the error?